lib/discovery_api/event_handler.ex

```
@@ -20,4 +22,29 @@ defmodule DiscoveryApi.EventHandler do
 
     :discard
   end
+
+  def handle_event(%Brook.Event{type: dataset_update(), data: %Dataset{} = dataset}) do
+    Logger.debug(fn -> "Handling dataset: `#{dataset.technical.systemName}`" end)
+
+    with {:ok, organization} <- DiscoveryApi.Schemas.Organizations.get_organization(dataset.technical.orgId),
+         {:ok, _cached} <- SystemNameCache.put(dataset, organization),
+         model <- Mapper.to_data_model(dataset, organization) do
+      DiscoveryApi.Search.Storage.index(model)
+      save_dataset_to_recommendation_engine(dataset)
+      ResponseCache.invalidate()
```

Does the order of these actions after the `do` matter?  And if they fail, will they raise or just fail silently?

---

After discussion - sounds like this was just lifted from the registry subscriber event listener.  All clear.